### ISSUE TITLE
Firebug 2.0

### ISSUE THREAD

**Issue by jennifer-shehane** [MikeRatcliffe/Acebug#58]
Would be great to see this add-on get updated to work with Firebug 2.0 !

**Comment by nightwing** [comment]
Hi, unfortunately firebug 2.0 have different implementation of code highlighting in the console, replacing which requires more work, and likely doesn't add enough value to keep Acebug popular. So most likely Acebug won't be updated.

**Comment by jennifer-shehane** [comment]
I'm really just interested in being able to write Coffeescript within the Firebug console. Would that be any easier to implement?

If not, maybe you can point me in the right direction to implement this myself.

**Comment by nightwing** [comment]
Yes adding support for simply evaluating coffeescript would be easier, basically requires changing `Firebug.CommandLine.enter` https://github.com/MikeRatcliffe/Acebug/blob/master/chrome/content/aceEditor.js#L395 and adding  a call to coffeescript compiler. https://github.com/MikeRatcliffe/Acebug/blob/master/chrome/content/aceEditor.js#L431-L442

Syntax highlighting for coffeescript, cells, error reporting require much more work, but if you are not interested in that, you can implement this easily.

**Comment by nightwing** [comment]
Looking at the new code, it might be easier to wrap `commandLine.evaluate` instead of overriding enter.  https://github.com/firebug/firebug/blob/master/extension/content/firebug/console/commandLine.js#L311.
Also have a look at https://github.com/firebug/extension-examples using restartless extension template like https://github.com/firebug/extension-examples/tree/master/HelloBootAMD might be easier than starting with old-style acebug.
For development instead of building you can create a text file named same as your addon id (`hellobootamd@janodvarko.cz`) with full path of your extension folder in it.
